lms/views/api/canvas/authorize.py

```diff
@@ -116,7 +116,7 @@ def oauth2_redirect(request):
 def oauth2_redirect_error(request):
     request.context.js_config.enable_oauth2_redirect_error_mode(
         auth_route="canvas_api.oauth.authorize",
-        error_details=request.params.get("error_description"),
+        error_details={"error_description": request.params.get("error_description")},
```

Comment:
This was the one place where a non-empty string was actually passed. Just turned it into a dict

Reply:
This change may change behavior on the front end because empty strings are falsy in JavaScript but empty objects are not. This could manifest in an "Error Details" section rendering but being empty of any real content. Is it possible to not return an empty dict (i.e. not have the property present)? Although it sounds like a step backwards, an empty string would be preferable here to an empty object (typing on the frontend can be taught to handle this).

Reply:
I changed it to just omit the `errorDetails` config setting entirely if there's no `error_description` request param from Canvas.